coordinatorlib/src/main/scala/com/socrata/datacoordinator/util/TimingReport.scala

```diff
@@ -69,6 +69,36 @@ class LoggedTimingReport(log: Logger) extends TimingReport {
   }
 }
 
+/**
+ * Put keys with prefix - tag: in MDC so that enclosing log statements can be tagged using %X{name} and log config.
+ * Example:
+ *   key -> tag:job-id
+ *   log4j.appender.console.props.layout.props.ConversionPattern -> "%X{job-id})"
+ */
+trait TaggableTimingReport extends TimingReport with Metrics {
+
+  private val Tag = "^tag:(.+)$".r
+
+  abstract override def apply[T](name: String, kv: (String, Any)*)(f: => T): T = {
+
+    val (regKv, tagKv) = kv.partition { case (tag, value) => Tag.findFirstIn(tag).isEmpty }
+
+    val tagK = tagKv.collect { case (tag, value) =>
```

Comment:
Could this be folded to `case (Tag(tag), value)`?  Or does the compiler not like that?  (I guess or do you want an error to be thrown if `tag` isn't a `Tag`.)